src/main/java/tickbot/ui/text/TextUi.java

```diff
@@ -0,0 +1,29 @@
+package tickbot.ui.text;
+
+import java.util.Scanner;
+
+import tickbot.ui.Output;
+import tickbot.ui.Parser;
+import tickbot.ui.Ui;
+
+/**
+ * The class to represent the text UI.
+ */
+public class TextUi implements Ui {
+    @Override
+    public void mainLoop(String[] args) {
+        Output.printMessage("Hello, this is tickbot! How can I help you?");
+        Scanner inputScanner = new Scanner(System.in);
+        Parser parser = new Parser();
+        boolean running = true;
+        while (running) {
+            System.out.print("==> ");
+            if (!inputScanner.hasNextLine()) {
+                break; // end of file
+            }
+            String command = inputScanner.nextLine();
+            running = parser.executeCommand(command);
```

Comment:
I like how you designed this part of the code. I feel that it is very unique for the method to return a boolean to tell the application to continue running and at the same time give the required outputs to the user.